eiffel-syntax-and-usage/the-links-object.md

```diff
@@ -0,0 +1,40 @@
+# The Links Object
+The __links__ object is an array of trace links to other Eiffel events. These trace links by definition always reference backwards in time – it is only possible to reference an event that has already occured. Each trace link is a tuple consisting of a type and a UUID corresponding to the __meta.id__ of the target event, on String format.
+
+Some link types allow multiple trace links, whereas others only allow one.
+
+Example syntax of a simple __links__ object:
+
+    "links": [
+      {"type": "CAUSE", "target": "8f3e0f94-5d11-46e7-ae02-91efa15d2329"},
+      {"type": "COMPOSITION", "target": "43ee71d2-6d91-496a-b9cf-d121ff1d1bcf"}
+    ]
+
+This section is an extension to the core Eiffel Protocol [Links Object](https://github.com/Ericsson/eiffel/blob/master/eiffel-syntax-and-usage/the-links-object.md).
+
+## Legal Link Types 
+### SERVICE
+__Required in:__ [EiffelServiceStoppedEvent](../eiffel-vocabulary/EiffelServiceStoppedEvent.md), [EiffelServicePausedEvent](../eiffel-vocabulary/EiffelServicePausedEvent.md), [EiffelServiceResumedEvent](../eiffel-vocabulary/EiffelServiceResumedEvent.md), [EiffelConfigurationOrderedEvent](../eiffel-vocabulary/EiffelConfigurationOrderedEvent.md), [EiffelConfigurationAppliedEvent](../eiffel-vocabulary/EiffelConfigurationAppliedEvent.md)   
+__Optional in:__ None  
+__Legal targets:__ [EiffelServiceStartedEvent](../eiffel-vocabulary/EiffelServiceStartedEvent.md)  
+__Multiple allowed:__ No  
+__Description:__ Identifies the service the event is targeting. 
+
+### CONFIGURATION_SOURCE
+ __Required in:__ None  
+ __Optional in:__ [EiffelConfigurationAppliedEvent](../eiffel-vocabulary/EiffelConfigurationAppliedEvent.md), [EiffelConfigurationOrderedEvent](../eiffel-vocabulary/EiffelConfigurationOrderedEvent.md)  
+ __Legal targets:__ [EiffelSourceChangeSubmittedEvent](https://github.com/Ericsson/eiffel/blob/master/eiffel-vocabulary/EiffelSourceChangeSubmittedEvent.md)  
+ __Multiple allowed:__ No  
+ __Description:__ Used to identify the configuration that has been applied to an item.
+
+## Extension additions to Legal Link Types in core Eiffel Protocol
+### ARTIFACT
+__Required in:__ [EiffelArtifactDeployedEvent](../eiffel-vocabulary/EiffelArtifactDeployedEvent.md)  
+__Optional in:__ None  
+__Legal targets:__ [EiffelArtifactCreatedEvent](https://github.com/Ericsson/eiffel/blob/master/eiffel-vocabulary/EiffelArtifactCreatedEvent.md)  
+
+### ENVIRONMENT
+__Required in:__ [EiffelServiceStartedEvent](../eiffel-vocabulary/EiffelServiceStartedEvent.md), [EiffelEnvironmentTerminatedEvent](../eiffel-vocabulary/EiffelEnvironmentTerminatedEvent.md)  
+__Optional in:__ None  
+__Legal targets:__ [EiffelEnvironmentDefinedEvent](https://github.com/Ericsson/eiffel/blob/master/eiffel-vocabulary/EiffelEnvironmentDefinedEvent.md)  
```

Comment:
Need some explanatory text. What does this ENVIRONMENT between these event types mean?